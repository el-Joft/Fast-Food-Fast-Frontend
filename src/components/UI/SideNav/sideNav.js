/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import SideNav from 'react-simple-sidenav';
import SideNavItem from './sideNavItem';
import NavItem from '../../NavItems/NavItems';

export const SideNavigation = props => (
  <div>
    <SideNav
      showNav={props.showNav}
      onHideNav={props.onHideNav}
      navStyle={{
        background: '#42ce9f',
        maxWidth: '220px',
      }}
    >
      <SideNavItem />
    </SideNav>
  </div>
);

export default SideNavigation;
