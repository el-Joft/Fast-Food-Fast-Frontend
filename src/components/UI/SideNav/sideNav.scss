.option {
  font-weight: 300!important;
  font-size: 12px!important;
  color: #ffffff!important;
  padding: 10px!important;
  display: block!important;
}

.option a {
  color: inherit;
  text-decoration: none;
}

.option span {
  color: #ffff;
  margin-right: 10px;
}
// .bars {
//   background: #42ce9f;
// }

@media only screen and (max-width: 768px) {
  .bars {
    background: #42ce9f;
  }
}