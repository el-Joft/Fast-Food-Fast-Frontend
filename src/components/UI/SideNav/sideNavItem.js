/* eslint-disable react/no-array-index-key */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/forbid-prop-types */
import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
// import './NavItems.scss';
import './sideNav.scss';
import { logoutUser } from '../../../actions/authActions';
import { addCart } from '../../../actions/orderAction';
import cartLogo from '../../../public/images/icons/cart.png';

export const SideNavItem = (props) => {
  const { isAuthenticated, user } = props.auth;
  let showItems = [];
  const items = [
    {
      text: 'Menu',
      link: '/menu',
    },
    {
      text: 'About Us',
      link: '/about',
    },
    {
      text: 'Login',
      link: '/login',
    },
    {
      text: 'Sign Up',
      link: '/signup',
    },
  ];
  const authenticated = [
    {
      text: 'Menu',
      link: '/menu',
    },
    {
      text: 'About Us',
      link: '/about',
    },
    {
      text: 'Welcome',
      link: '/dashboard',
    },
  ];

  const cart = () => {
    const cartItem = localStorage.getItem('cart');
    let cartLength = 0;
    if (cartItem) {
      const parsedItem = JSON.parse(cartItem);
      cartLength = parsedItem.length;
    }
    return cartLength;
  };

  if (isAuthenticated) {
    showItems = () => authenticated.map((item, i) => (
      <li key={i}>
        <Link
          to={item.link}
        >
          {item.text}
        </Link>
      </li>
    ));
  } else {
    showItems = () => items.map((item, i) => (
      <li key={i}>
        <Link to={item.link}>
          {item.text}
        </Link>
      </li>
    ));
  }

  const logoutClick = (e) => {
    e.preventDefault();
    props.logoutUser();
  };


  return (
    <div>
      <nav>
        <ul className="nav-bar option">
          {showItems()}
          { isAuthenticated
            ? (
              <li>
                <Link
                  onClick={e => logoutClick(e)}
                  to='/logout'
                >
                  Logout
                </Link>
              </li>
            ) : ''
          }
          <li className="cart-section">
            <Link to="/checkout">
              <img src={cartLogo} alt="" />
              <span>
Cart(
                {cart()}
)
              </span>
            </Link>
          </li>
        </ul>
      </nav>
    </div>
  );
};

SideNavItem.propTypes = {
  logoutUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  auth: state.auth,
  order: state.order,
});

export default connect(mapStateToProps, { logoutUser, addCart })(SideNavItem);
