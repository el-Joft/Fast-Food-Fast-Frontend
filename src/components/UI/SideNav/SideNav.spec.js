import React from 'react';
import { shallow } from 'enzyme';
import SideNav from 'react-simple-sidenav';
import { SideNavigation } from './sideNav';

describe.only('shallow testing', () => {
  let wrapper;
  it('should test the component if it renders anything', () => {
    wrapper = shallow(<SideNavigation />);
    expect(wrapper.length).toEqual(1);
  });

  it('should test the component if it renders anything', () => {
    wrapper = shallow(<SideNavigation />);
    const li = wrapper.find(SideNav);
    expect(li.length).toEqual(1);
  });

  it('should test the component if it renders anything', () => {
    wrapper = shallow(<SideNavigation />);
    expect(wrapper).toMatchSnapshot();
  });
});
