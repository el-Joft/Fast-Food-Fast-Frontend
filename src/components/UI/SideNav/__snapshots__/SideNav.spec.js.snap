// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow testing should test the component if it renders anything 1`] = `
<div>
  <e
    navStyle={
      Object {
        "background": "#42ce9f",
        "maxWidth": "220px",
      }
    }
  >
    <Connect(SideNavItem) />
  </e>
</div>
`;
